### ISSUE TITLE
Is there a way to fail docker:start if one of the contains has a certain log output?

### ISSUE THREAD

**Issue by mason** [rhuss/docker-maven-plugin#252]
I am trying to run tests inside of a running container and when the build is successful maven prints out "BUILD SUCCESSFUL" and if it fails it prints "BUILD FAILURE". Right now I am using the log of "BUILD SUCCESSFUL" to mark the starting of the container as successful. But there is no option to fail mvn docker:start if the tests fail and the logs show "BUILD FAILURE". The only thing I can do it have a timeout and have it fail that way. Any ideas?

**Comment by jgangemi** [comment]
this isn't currently possible but could be added as an enhancement.

**Comment by mason** [comment]
@jgangemi that would be an awesome enhancement! :+1:

**Comment by rhuss** [comment]
:+1: from my side, too. I think it would be a valuable addition to have something in the `<wait>` section which will break the build (something like `<fail>` or `<breakIf>` or `<failLog>` or ...)

**Comment by jgangemi** [comment]
i was thinking of going w/ a `fail` element that worked in a similar manner to `log`.

**Comment by rhuss** [comment]
sounds good, go ahead ! ;-)

**Comment by mgurov** [comment]
How about wrapping normal wait conditions (`log` and/or `http`) with `not` ? e.g. 

````xml
                <wait>
                  <log>BUILD SUCCESSFUL</log>
                  <not>
                    <log>BUILD FAILURE</log>
                  </not>
                </wait>
````

IMHO clear enough and leaves the door open for more elaborate configuration if we want later support `<and>`, `<or>` with nested negations.

**Comment by rhuss** [comment]
You are right, that would be probably more flexible. However TBH I don't intend to create some sort of DSL only for the wait conditions. For logs, IMO its more straight forward to have `<log>` and `<fail>`, for `<http>` we also have for the next version the possibility to specify a range of status code which is supposed to full fill the wait condition. /wr to `<and>` and `<or>` I don't currently see a use case where this is required (Do you have one ?)

Also `<wait><not><log>` is IMO not so super intuitive, because for me it reads like *wait until the log entry is not there* ? 

I would wait until we really need more complex wait condition in which case we might to allow a real DSL (e.g. sth groovy based instead of trying to script via XML which is really tedious to read and maintain).

**Comment by mgurov** [comment]
Well, I guess earlier or later you might want to wait for first of two pattern lines appearing in the log, or both not appearing. But I agree that the simple `<fail>` would cover most cases.

I might pick this up if @jgangemi has a delay.

**Comment by mgurov** [comment]
I have just encountered a case where `<and>` would be nice to have. I use a small suite of "integration" poms to test the changes I do for this plugin (`docker-maven-plugin`). This same plugin is used to execute those integration poms against [maven docker image](https://hub.docker.com/r/library/maven/),  and sometimes it would be nicer to be able to wait for multiple log patterns. 

Take [this example](https://github.com/mgurov/docker-maven-runs/blob/noopWait/pom.xml#L138), where the changes for the `<wait/>` clause are tested. It would be better to check not only that the wait output is such and such, but also that the maven build succeeds or fails.

I would agree though that this example is somewhat marginal yet and doesn't justify introduction of a heavy DSL.
